### PR TITLE
CDAP-17337 support configuring datasetName

### PR DESCRIPTION
**Pull request by seanzhougoogle**

### CODE REVIEW

File: src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java [hunk 1]
```diff
@@ -192,9 +193,16 @@
   // cannot write to a temporary file on local disk either in case there is a failure writing to disk
   // Without keeping the entire batch in memory, there would be no way to recover the records that failed to write
 
+  BigQueryEventConsumer(DeltaTargetContext context, Storage storage, BigQuery bigQuery, Bucket bucket,
+    String project, int loadIntervalSeconds, String stagingTablePrefix, boolean requireManualDrops,
+    @Nullable EncryptionConfiguration encryptionConfig, @Nullable Long baseRetryDelay) {
+    this(context, storage, bigQuery, bucket, project, loadIntervalSeconds, stagingTablePrefix, requireManualDrops,
+      encryptionConfig, baseRetryDelay, null);
+  }
```
Comment (sagarkapare) [inline src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java]: nit: why not add `datasetName` to the existing constructor?

File: widgets/bigquery-cdcTarget.json [hunk 2]
```diff
@@ -33,6 +33,11 @@
     {
       "label" : "Advanced",
       "properties" : [
+        {
+          "name": "datasetName",
```
Comment (sagarkapare) [inline widgets/bigquery-cdcTarget.json:37]: Let's add this field info the the documentation as well i.e. `md` file.

File: src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java [hunk 3]
```diff
@@ -186,6 +186,14 @@ private static String stringifyPipelineId(DeltaPipelineId pipelineId) {
     @Description("Whether to require manual intervention when a drop table or drop database event is encountered.")
     private Boolean requireManualDrops;
 
+    @Nullable
+    @Description("Optional. By default the dataset name is same as source database name.")
```
Comment (sagarkapare) [inline src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java]: Would be good to document the naming convention that need to be followed. For example : `valid name should only contain letters , numbers, and underscores and maximum length can be 1024 chars. Any invalid chars would be replaced by _ in the final dataset name`

File: src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java (hunk 1 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java]: done:)

File: widgets/bigquery-cdcTarget.json (hunk 2 above)
Comment (seanzhougoogle) [reply widgets/bigquery-cdcTarget.json:37]: done :)

File: src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java (hunk 3 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/bigquery/BigQueryTarget.java]: good point :)

File: docs/bigquery-cdcTarget.md [hunk 4]
```diff
@@ -65,4 +65,10 @@ Staging tables names are generated by prepending this prefix to the target table
 datasets when a drop table or drop database event is encountered. When set to true, the replicator will
 not delete a table or dataset. Instead, it will fail and retry until the table or dataset does not exist.
 If the dataset or table does not already exist, no manual intervention is required. The event will be
-skipped as normal.
\ No newline at end of file
+skipped as normal.
+
+**Dataset Name**: Name of the dataset to be created in the BQ query. It's optional and by default the dataset
```
Comment (sagarkapare) [inline docs/bigquery-cdcTarget.md]: minor nit: `created in the BigQuery.` instead of `created in the BQ query.`

**Review by sagarkapare — APPROVED** [review]

File: docs/bigquery-cdcTarget.md (hunk 4 above)
Comment (seanzhougoogle) [reply docs/bigquery-cdcTarget.md]: good catch :)

**Pull request merged**